### ISSUE TITLE
/search endpoint support

### ISSUE THREAD

**Issue by brettgerry** [nylas/nylas-nodejs#22]
Support search on Threads and Messages
https://nylas.com/docs/platform#search

**Comment by mhwheeler** [comment]
+1

**Comment by mccolin** [comment]
Definitely would be nice to have official lib support for search. If you need search, like yesterday, you can make a direct request to the search endpoint:

```
var api = Nylas.with(accessToken);
api.request({
  method:'GET', 
  path: '/messages/search', 
  qs: {q: "Your Gmail-, Outlook-, Yahoo-, or IMAP-style query here"
}}).then( function(results) {
  console.log(results.length, "search results returned:", results);
});
```

**Comment by jackiehluo** [comment]
This feature landed in https://github.com/nylas/nylas-nodejs/commit/d47f9bbbb4525feecdfafbc1fe59aa62d78088ef! It'll be in the next release.

**Comment by jackiehluo** [comment]
Fixed in 4.0.0!
